### PR TITLE
Add pydantic-ai example

### PR DESCRIPTION
**Pull request by MarcSkovMadsen**
I created this example to try out [pydantic-ai](https://ai.pydantic.dev/). Then I thought it was worth persisting here.

I cannot install llama-cpp in my development environent. I don't believe I can install linux dependencies to generate screenshots and videos in my environment. Thus I created thumbnail and video manually. Feel free to create via commands.

https://github.com/user-attachments/assets/b5fa2326-9ce0-49e7-bc7e-f617b007ed7b

pydantic-ai looks nice. For me the main downside is the very strong integration with logfire for telemetry. Would be very nice to display telemetry directly in the Panel app instead for the internal tools I create.

### CODE REVIEW

File: docs/applicable_recipes.md [hunk 1]
```diff
@@ -1,105 +1,6 @@
...
-template = pn.template.BootstrapTemplate(
-    title="OpenAI DALL·E",
-    header_background="#212121",
-    main=[chat_interface],
-    header=[api_key_input],
-    sidebar=[model_buttons, size_buttons, n_images_slider],
-)
-template.servable()
-```
-</details>
```
Comment (ahuang11) [inline docs/applicable_recipes.md:100]: Was this intentional?

Comment (ahuang11) [reply docs/applicable_recipes.md:100]: Nevermind I see it moved below

Comment (MarcSkovMadsen) [reply docs/applicable_recipes.md:100]: If what was intentional?

**Review by ahuang11 — APPROVED** [review]

**Comment by MarcSkovMadsen** [comment]
> I think just linting; otherwise looks good!

Done. I don't know why pre-commit run --all locally did not fix lint issues?

**Comment by ahuang11** [comment]
It seems like there's an issue with the lint setup


cmd [1] | isort . --check-only
Skipped 1 files
cmd [2] | black --check .
All done! ✨ 🍰 ✨
29 files would be left unchanged.
cmd [3] | ruff check .
warning: The top-level linter settings are deprecated in favour of their counterparts in the `lint` section. Please update the following options in `pyproject.toml`:
  - 'extend-select' -> 'lint.extend-select'
docs/examples/applicable_recipes/pydanticai_find_city_agent.py:11:8: F401 [*] `param` imported but unused

**Comment by MarcSkovMadsen** [comment]
Problems seems fixed. Should I merge?

**Comment by ahuang11** [comment]
Thanks! Merged!

**Comment by MarcSkovMadsen** [comment]
Thanks. Do you publish this? Update the website?

I'm not sure I know how to do this.

**Comment by ahuang11** [comment]
Should be automated on merge. See actions

**Comment by MarcSkovMadsen** [comment]
Seems like the docs html is not rebuilt as the released version is not fully up to date:

image

**Comment by ahuang11** [comment]
Can you clarify? That is the text there:
https://github.com/holoviz-topics/panel-chat-examples/pull/152/files#diff-a233a2c1408611781b6b400257bb72154b88ea78629e3d21e333a23055233b09R146

**Comment by MarcSkovMadsen** [comment]
I would expect to see what is shown on the left. But when I run build below in terminal it changes to the older diff on the right. Don't know?

image

**Comment by MarcSkovMadsen** [comment]
Ahh. Now I know. Its because the markdown is autogenerated from the .py file. Did not remember that.

**Comment by ahuang11** [comment]
It's based on the docstring.

**Comment by ahuang11** [comment]
Yeah I forgot about that too. Maybe we should leave a comment up top saying `# DO NOT EDIT`

**Pull request merged**